File: 78-subsets/main.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        res = []
        
        def dfs(i: int, path: List[int]) -> None:
            if i == n:
                res.append(path)
                return
            
            dfs(i+1, path)
            dfs(i+1, path + [nums[i]])
        
        dfs(0, [])
        return res

File: 78-subsets/main.go
package main

import (
	"sort"
)

func subsets(nums []int) [][]int {

	sort.Ints(nums)
	var res [][]int
	var backtrack func(int, []int)

	backtrack = func(start int, path []int) {
		tmp := make([]int, len(path))
		copy(tmp, path)
		res = append(res, tmp)
		for i := start; i < len(nums); i++ {
			path = append(path, nums[i])
			backtrack(i+1, path)
			path = path[:len(path)-1]
		}
	}
	backtrack(0, []int{})
	return res
}
